sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/DeltaMetricStorage.java

```diff
@@ -47,9 +53,13 @@ public BoundStorageHandle bind(Attributes attributes) {
       return aggregatorHandle;
     }
 
-    // Missing entry or no longer mapped, try to add a new entry.
+    // Missing entry or no longer mapped. Try to add a new one if not exceeded cardinality limits.
     aggregatorHandle = aggregator.createHandle();
     while (true) {
+      if (activeCollectionStorage.size() >= MAX_ACCUMULATIONS) {
+        // TODO: provide useful diagnostics when max accumulations has been reached
+        return NOOP_STORAGE_HANDLE;
```

Comment:
I def understand not wanting to fill up the logs when this condition is hit, but I do feel like especially while we're in alpha (and probably even later) it's important to have SOME indication that data is being dropped. How about logging max once a minute or maybe just a one-shot?

Reply:
Yeah I think it's important too. I like logging once when the limit is exceeded, or perhaps every 10th time? I.e. something like:
```
AtomicLong exceedsMaxAccumulationsCount = new AtomicLong();
...
if (activeCollectionStorage.size() >= MAX_ACCUMULATIONS) {
       if (exceedsMaxAccumulationsCount.getAndIncrement() % 10 == 0) {
          logger.log("Max accumulations exceeded for instrument ....");
       }
       return NOOP_STORAGE_HANDLE;
}
```

Reply:
@jkwatson has a ThrottlingLogger you can use here.  Please do!